pageserver/src/context.rs

```diff
@@ -0,0 +1,198 @@
+//! This module defines `RequestContext`, a structure that we use throughout
+//! the pageserver codebase to propagate high-level context from the places
+//! that _originate_ activity down to the shared code paths at the
+//! heart of the pageserver.
+//!
+//! For example, the initial product needs were that we needed to answer the
+//! following questions inside `Timline::get(page_nr, lsn)`:
+//! 1. What high-level activity ([`TaskKind`]) needs this page?
+//!    We need that information as a categorical dimension for page access
+//!    statistics, which we, in turn, need to guide layer eviction policy design.
+//! 2. How should we behave if, to produce the page image, we need to
+//!    on-demand download a layer file ([`DownloadBehavior`]).
+//!
+//! [`RequestContext`] satisfies those needs.
+//! The current implementation is a small `struct` that is passed through
+//! the call chain by reference.
+//!
```

Comment:
All the text that follows concerns things that are not implemented here yet. When I read this from beginning to end, I didn't realize that until I reached the "We do not yet have a systematic cancellation story" paragraph. Let's add here:

```suggestion
//!
//! # TODO
//!
```